src/workerd/api/cache.c++

```diff
@@ -504,13 +513,8 @@ kj::Own<kj::HttpClient> Cache::getHttpClient(IoContext& context,
                                              kj::StringPtr operationName) {
   auto span = context.makeTraceSpan(operationName);
 
-  auto cacheClient = context.getCacheClient();
-  auto httpClient = cacheName.map([&](kj::String& n) {
-    return cacheClient->getNamespace(n, kj::mv(cfBlobJson), span);
-  }).orDefault([&]() {
-    return cacheClient->getDefault(kj::mv(cfBlobJson), span);
-  });
-  httpClient = httpClient.attach(kj::mv(span), kj::mv(cacheClient));
+  auto httpClient = context.getHttpClient(IoContext::CACHE_CHANNEL, true, kj::mv(cfBlobJson), operationName);
```

Comment:
Sorry, this isn't the right approach. The code here should not have to change. Instead you should be implementing `IoChannelFactory::getCache()` in `server.c++`. The only changes that should be needed in this PR are to `server.c++` and `worker.capnp`.

Reply:
Ah, I see. How should the `getCache` implementation behave? Internally it looks like it's an http client, would that work? Is the approach of defining a special subrequest channel along the right lines?